Parser/tokenizer.h

```diff
@@ -60,8 +60,8 @@ struct tok_state {
     PyObject *decoding_readline; /* open(...).readline */
     PyObject *decoding_buffer;
     const char* enc;        /* Encoding for the current str. */
-    const char* str;
-    const char* input; /* Tokenizer's newline translated copy of the string. */
```

Comment:
What's the reason these can't be const?

Reply:
`tok->input` can't be const because it's the allocated input, and then gets freed in `PyTokenizer__Free`.

`tok->str` can't be const because it can be returned from `decode_str`, which returns a non-const string.  If I inlined `decode_str` into `PyTokenizer_FromString`, the only place that uses it, then I think I could make `tok->str` const.

Reply:
`decode_str` only returns a non-const string in this change, though.

Reply:
That's correct. I changed `decode_str` to return `char *` instead of `const char *` because it can return the result from `translate_newlines`.  Also, the result from `decode_str` is assigned to `tok->buf` which is non-const.